docs/writing-tests/wdspec.md

```diff
@@ -66,3 +67,97 @@ a WebDriver session) are defined in terms of Pytest "fixtures" and must be
 loaded accordingly. For more detail on how to define and use test fixtures,
 please refer to [the pytest project's documentation on the
 topic](https://docs.pytest.org/en/latest/fixture.html).
+
+## WebDriver BiDi
+
+The wdspec tests for [WebDriver BiDi](https://w3c.github.io/webdriver-bidi) are
+located in the `tests/bidi/` and `tests/interop` directories. Tests related to
+external specifications (like [Permissions](https://www.w3.org/TR/permissions/) or
+[Web Bluetooth](https://webbluetoothcg.github.io/web-bluetooth)) are located in
+`external` subdirectories. E.g. `tests/bidi/external/bluetooth/`.
+
+The `webdriver.bidi.client.BidiSession` class provides an abstraction for the BiDi
+client and contains properties corresponding to the
+[WebDriver BiDi modules](https://w3c.github.io/webdriver-bidi/#protocol-modules). It
+can be retrieved by fixture `bidi_session`.
+
+### Extending WebDriver BiDi
+
+This section describes how to extend the WebDriver BiDi client with an example of
+adding support for [Web Bluetooth](https://webbluetoothcg.github.io/web-bluetooth).
+
+#### Adding a New Module
+
+##### Create `BidiModule`
+
+BiDi modules are defined in the `tools/webdriver/webdriver/bidi/modules/` directory.
+To add a new module called `bluetooth`, declare a Python class
```

Comment:
I would quite strongly prefer the examples here to be something that has multi-vendor support.

Reply:
I would agree here. The `Permissions` API can be used instead for such an example. It's one which has cross-browser support.

Reply:
I am not sure this guide requires a specific module as an example, I would be happy with a `foo` module using a `bar` command but permissions sounds good to me too.

Reply:
Switched to `Permissions`, as it allows to provide a specific code snippets.

Reply:
@jgraham was it the only blocker for the spec?